### ISSUE TITLE
Make "Open Recent" as sub-menu of menu "File" (+ add option to "Clear Recent")

### ISSUE THREAD

**Issue by Symbian9** [leozide/leocad#216]
### Actual behavior

Currently list of recently used files shown directly in the root of menu *"File"*, where long path of file could spread this menu too much
![LeoCAD Recent]http://pasteall.org/pic/show.php?id=135877

### Expected behavior

- All recently used files should be shown optionally only if user will open sub-menu *"Open Recent"*.
- There should be option *"Clear Recent"* for clear this list.

### Reference

sK1 2.0
![sK1 Recent](http://pasteall.org/pic/show.php?id=135879)

AzPainter
![AzPainter Recent](http://pasteall.org/pic/show.php?id=135881)

**Comment by leozide** [comment]
I don't like the submenu idea, it adds 1 extra click to my workflow and breaks quickly pressing alt+f 1 to open a recent file. I'd prefer if the long file names were just elided.

**Comment by Symbian9** [comment]
@leozide , think, as cmpromise solutiion add option "Show list of Recent `N` files" (where `N` is drop-down list with values from 1 to 9) to "Preferences > General"

![pic.1](https://i.imgur.com/tfNiRvN.png)

**Comment by Symbian9** [comment]
For quick fix this issue, please, add checkbox option ***"Hide list of recent files"*** in *"Preferences > General"* tab.

Think, this would be the best solution here.

**Comment by Symbian9** [comment]
> For quick fix this issue, please, add checkbox option ***"Hide list of recent files"*** in *"Preferences > General"* tab.

@leozide, could you add this option?

**Comment by Symbian9** [comment]
> I'd prefer if the long file names were just elided.

@leozide This is really issue which should be fixed.

> I don't like the sub-menu idea, it adds 1 extra click

BTW, I still propose add *"Open Recent..."* as sub-entry in *"File"* menu, as it is already de-facto standard in desktop apps UI/UX (GIMP, Inkscape, QCAD, FreeCAD and all other most popular FLOSS apps).

Otherwise add at least check box option to disable for show recent files, for those AFOLs which has no need to keep recent list in "File" menu.

![pic.1](https://i.imgur.com/PfnI6nY.png)

![pic.2](https://i.imgur.com/iBDkzHK.png)

![pic.3](https://i.imgur.com/ML00CgO.png)

**Comment by Lswbanban** [comment]
Yes, the "Open Recent" is what I was asking (which I called a sub menu of the File menu).

I agree it is kind of a standard in Desktop application, and I'm not convinced with the argument saying it would require one more click (yes it does, but no big deal, as it is not an operation you do hundreds of time when using the software)...
